x/ibc/02-client/types/tendermint/misbehaviour.go

```diff
@@ -1,96 +1,48 @@
-
-	sdk "github.com/cosmos/cosmos-sdk/types"
+	evidenceexported "github.com/cosmos/cosmos-sdk/x/evidence/exported"
 	"github.com/cosmos/cosmos-sdk/x/ibc/02-client/exported"
 	"github.com/cosmos/cosmos-sdk/x/ibc/02-client/types/errors"
-
-	"github.com/tendermint/tendermint/crypto/tmhash"
-	cmn "github.com/tendermint/tendermint/libs/common"
-	tmtypes "github.com/tendermint/tendermint/types"
+	host "github.com/cosmos/cosmos-sdk/x/ibc/24-host"
 )
 
-var _ exported.Evidence = Evidence{}
+var _ exported.Misbehaviour = Misbehaviour{}
+var _ evidenceexported.Evidence = Misbehaviour{}
 
-// Evidence is a wrapper over tendermint's DuplicateVoteEvidence
-// that implements Evidence interface expected by ICS-02
-type Evidence struct {
-	*tmtypes.DuplicateVoteEvidence
-	ChainID        string `json:"chain_id" yaml:"chain_id"`
-	ValidatorPower int64  `json:"val_power" yaml:"val_power"`
-	TotalPower     int64  `json:"total_power" yaml:"total_power"`
+// Misbehaviour contains an evidence that a
+type Misbehaviour struct {
+	*Evidence
+	ClientID string `json:"client_id" yaml:"client_id"`
 }
 
-// Type implements exported.Evidence interface
-func (ev Evidence) Route() string {
-	return exported.ClientTypeTendermint
+// ClientType is Tendermint light client
+func (m Misbehaviour) ClientType() exported.ClientType {
+	return exported.Tendermint
 }
 
-// Type implements exported.Evidence interface
-func (ev Evidence) Type() string {
-	return exported.ClientTypeTendermint
+// GetEvidence returns the evidence to handle a light client misbehaviour
+func (m Misbehaviour) GetEvidence() evidenceexported.Evidence {
+	return m.Evidence
 }
 
-// String implements exported.Evidence interface
-func (ev Evidence) String() string {
-	bz, err := yaml.Marshal(ev)
-	if err != nil {
-		panic(err)
+// ValidateBasic performs the basic validity checks for the evidence and the
+// client ID.
+func (m Misbehaviour) ValidateBasic() error {
+	if m.Evidence == nil {
+		return errors.ErrInvalidEvidence(errors.DefaultCodespace, "empty evidence")
 	}
-	return string(bz)
-}
 
-// Hash implements exported.Evidence interface
-func (ev Evidence) Hash() cmn.HexBytes {
-	return tmhash.Sum(SubModuleCdc.MustMarshalBinaryBare(ev))
-}
-
-// ValidateBasic implements exported.Evidence interface
-func (ev Evidence) ValidateBasic() sdk.Error {
-	if ev.DuplicateVoteEvidence == nil {
-		return sdk.ConvertError(errors.ErrInvalidEvidence(errors.DefaultCodespace, "duplicate evidence is nil"))
-	}
-	err := ev.DuplicateVoteEvidence.ValidateBasic()
-	if err != nil {
-		return sdk.ConvertError(errors.ErrInvalidEvidence(errors.DefaultCodespace, err.Error()))
-	}
-	if ev.ChainID == "" {
-		return sdk.ConvertError(errors.ErrInvalidEvidence(errors.DefaultCodespace, "chainID is empty"))
-	}
-	if ev.ValidatorPower <= 0 {
-		return sdk.ConvertError(errors.ErrInvalidEvidence(errors.DefaultCodespace, fmt.Sprintf("Invalid Validator Power: %d", ev.ValidatorPower)))
+	if err := m.Evidence.ValidateBasic(); err != nil {
```

Comment:
needs to also check `m.ChainID == m.Evidence.ChainID`

Reply:
see comment above